tests/integration/helm/install_test.go

```diff
@@ -0,0 +1,218 @@
+	IngressReleaseName  = IngressGatewayChart
+	EgressReleaseName   = EgressGatewayChart
+	ControlChartsDir    = "istio-control"
+	GatewayChartsDir    = "gateways"
+	retryDelay          = 2 * time.Second
+	retryTimeOut        = 20 * time.Minute
+)
+
+var (
+	// ChartPath is path of local Helm charts used for testing.
+	ChartPath = filepath.Join(env.IstioSrc, "manifests/charts")
+)
+
+// TestDefaultInstall tests Istio installation using Helm with default options
+func TestDefaultInstall(t *testing.T) {
+	framework.
+		NewTest(t).
+		Run(func(ctx framework.TestContext) {
+			workDir, err := ctx.CreateTmpDirectory("helm-install-test")
+			if err != nil {
+				t.Fatal("failed to create test directory")
+			}
+			cs := ctx.Environment().(*kube.Environment).KubeClusters[0]
+			h := helm.New(cs.Filename(), ChartPath)
+			s, err := image.SettingsFromCommandLine()
+			if err != nil {
+				t.Fatal(err)
+			}
+			overrideValuesStr := `
+global:
+  hub: %s
+  tag: %s
+`
+			overrideValues := fmt.Sprintf(overrideValuesStr, s.Hub, s.Tag)
+			overrideValuesFile := filepath.Join(workDir, "values.yaml")
+			if err := ioutil.WriteFile(overrideValuesFile, []byte(overrideValues), os.ModePerm); err != nil {
+				t.Fatalf("failed to write iop cr file: %v", err)
+			}
+			installIstio(t, ctx, cs, h, overrideValuesFile)
+
+			verifyInstallation(t, ctx, cs)
+
+			t.Cleanup(func() {
+				deleteIstio(t, cs, h)
+			})
+		})
+}
+
+// TestInstallWithFirstPartyJwt tests Istio installation using Helm
+// with first-party-jwt enabled
+func TestInstallWithFirstPartyJwt(t *testing.T) {
+	framework.
+		NewTest(t).
+		Run(func(ctx framework.TestContext) {
+			workDir, err := ctx.CreateTmpDirectory("helm-install-test")
+			if err != nil {
+				t.Fatal("failed to create test directory")
+			}
+			cs := ctx.Environment().(*kube.Environment).KubeClusters[0]
+			h := helm.New(cs.Filename(), ChartPath)
+			s, err := image.SettingsFromCommandLine()
+			if err != nil {
+				t.Fatal(err)
+			}
+			overrideValuesStr := `
+global:
+  hub: %s
+  tag: %s
+  jwtPolicy: first-party-jwt
+`
+			overrideValues := fmt.Sprintf(overrideValuesStr, s.Hub, s.Tag)
+			overrideValuesFile := filepath.Join(workDir, "values.yaml")
+			if err := ioutil.WriteFile(overrideValuesFile, []byte(overrideValues), os.ModePerm); err != nil {
+				t.Fatalf("failed to write iop cr file: %v", err)
+			}
+			installIstio(t, ctx, cs, h, overrideValuesFile)
+
+			verifyInstallation(t, ctx, cs)
+
+			t.Cleanup(func() {
+				deleteIstio(t, cs, h)
+			})
+		})
+}
+
+// installIstio install Istio using Helm charts with the provided
+// override values file and fails the tests on any failures.
+func installIstio(t *testing.T, ctx resource.Context, cs resource.Cluster,
+	h *helm.Helm, overrideValuesFile string) {
+	if _, err := cs.CoreV1().Namespaces().Create(context.TODO(), &kubeApiCore.Namespace{
+		ObjectMeta: kubeApiMeta.ObjectMeta{
+			Name: IstioNamespace,
+		},
+	}, kubeApiMeta.CreateOptions{}); err != nil {
```

Comment:
nit: you can use kerrors.isAlreadyExists() instead of another Get

Reply:
Done